dataland-frontend/src/utils/smoothScroll.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Smooth scroling
+ *
+ * @param target - target element
+ */
+export function smoothScroll(target: string): void {
+  const targetElement = document.querySelector(target) as HTMLElement;
+  const targetPosition = targetElement.getBoundingClientRect().top + window.scrollY - 100;
+  const startPosition = window.scrollY;
+  const distance = targetPosition - startPosition;
+  const duration = 300;
+  let startTime = null as unknown as number;
+
+  const animation = (currentTime: number): void => {
+    if (!startTime) {
+      startTime = currentTime;
+    }
```

Comment:
This will never be executed because `startTime` is always `null` at this point. Isn't it?

Reply:
It will only triggered once at the beginning when
startTime = null in the next call to startTime will have the value from currentTime which will be passed by the browser

Reply:
Ah ok so the "animation" callback is triggered multiple times during the display of the animation?

Sorry I did not know that since it is the first time I read animation code :) 

Just another idea - and correct me if it makes no sense - :
Could you then define the `amimation()` callback outside of `smoothScroll` ?
Because `smoothScroll()` is triggered each time someone clicks on a jump link, so the animation callback is declared multiple times from scratch.

You could pass the required params as inputs (e.g. startPosition, targetPosition, distance etc.)

Reply:
Yes. Good Idea